functions/New-DbaDbSequence.ps1

```diff
@@ -0,0 +1,190 @@
+            }
+
+            if ($Pscmdlet.ShouldProcess($db.Parent.Name, "Creating the sequence $Name in the $Schema schema in the database $($db.Name) on $($db.Parent.Name)")) {
+                try {
+                    if (Test-Bound Schema) {
+                        $schemaObject = $db | Get-DbaDbSchema -Schema $Schema
+
+                        # if the schema does not exist then create it
+                        if ($null -eq $schemaObject) {
+                            $schemaObject = $db | New-DbaDbSchema -Schema $Schema
+                        }
+                    }
+
+                    $newSequence = New-Object Microsoft.SqlServer.Management.Smo.Sequence -ArgumentList $db, $Name, $Schema
+                    $newSequence.IncrementValue = $IncrementBy
+                    $newSequence.IsCycleEnabled = $Cycle.IsPresent
+
+                    # support for user defined integer types
+                    $tableParts = $IntegerType -Split "\."
+
+                    if ($db.UserDefinedDataTypes[$IntegerType]) {
+                        # check to see if the type is in the user defined types for this db
+                        $newSequence.DataType = $db.UserDefinedDataTypes[$IntegerType]
+                    } elseif ($tableParts.Count -eq 2) {
+                        # custom type with the format "schema.typename"
+                        $newSequence.DataType = $db.UserDefinedDataTypes | Where-Object { $_.Schema -eq $tableParts[0] -and $_.Name -eq $tableParts[1] }
+                    } else {
+                        # system integer type
+                        $newSequence.DataType = New-Object Microsoft.SqlServer.Management.Smo.DataType $IntegerType
+                    }
+
+                    if (Test-Bound StartWith) {
+                        $newSequence.StartValue = $StartWith
+                    }
+
+                    if (Test-Bound MinValue) {
+                        $newSequence.MinValue = $MinValue
+                    }
+
+                    if (Test-Bound MaxValue) {
+                        $newSequence.MaxValue = $MaxValue
+                    }
+
+                    if (Test-Bound CacheSize) {
+                        if ($CacheSize -eq 0) {
+                            $newSequence.SequenceCacheType = [Microsoft.SqlServer.Management.Smo.SequenceCacheType]::NoCache
+                        } else {
+                            $newSequence.SequenceCacheType = [Microsoft.SqlServer.Management.Smo.SequenceCacheType]::CacheWithSize
+                            $newSequence.CacheSize = $CacheSize
+                        }
+                    } else {
+                        $newSequence.SequenceCacheType = [Microsoft.SqlServer.Management.Smo.SequenceCacheType]::DefaultCache
+                    }
+
+                    $newSequence.Create()
+                    $newSequence
```

Comment:
This should reach back to the instance (using the same connection as used to create it) and grab the sequence just created instead of depending on the cached object. 

Execution `Create()` will not always give good enough indicator that the object was indeed created.

Reply:
ok, will do